{
  try {
    this.session.save();
  }
 catch (  InvalidItemStateException e) {
    String message="Your operation was failed because it conflicts with a change made through another user. Please try again.";
    log.error("Caught Exception",e);
    throw new RulesRepositoryException(message,e);
  }
catch (  Exception e) {
    if (e instanceof RuntimeException) {
      throw (RuntimeException)e;
    }
 else {
      throw new RulesRepositoryException(e);
    }
  }
}
